### PR TITLE
integration tests

### PR DESCRIPTION
**Pull request by lory87**
**This pull request fixes the following issues**:

- adds testing setup and infrastructure for django 1.8 and 1.9
- adds basic test for admin style
- adds integration tests to the docs
- adds travis
- adds linting

### CODE REVIEW

File: djangocms_admin_style/tests/integration/helpers/cms.js [hunk 1]
```diff
@@ -0,0 +1,607 @@
+/* global document, localStorage */
+'use strict';
+
+module.exports = function (casperjs, settings) {
```
Comment (vxsx) [inline djangocms_admin_style/tests/integration/helpers/cms.js]: just as a reminder - these helpers will only work in develop (some stuff is different from 3.2.x)
also we will need to find a way to publish it somehow

File: djangocms_admin_style/tests/casperjs.conf.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            onPass: function (test) {
+                casper.test.pass('No changes found for screenshot ' + test.filename);
+            },
+            onComplete: function (allTests, noOfFails, noOfErrors) {
+                allTests.forEach(function (test) {
+                    if (test.fail) {
+                        console.log(test.filename, test.mismatch);
+                    }
+                });
+            },
```
Comment (vxsx) [inline djangocms_admin_style/tests/casperjs.conf.js]: missing https://github.com/divio/django-cms/compare/develop...FinalAngel:feature/integration-tests/screenshots#diff-3128fc90c810d166165d19095a68b502R39 fileNameGetter

File: testserver.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        'aldryn_bootstrap3',
+        'cms.test_utils.project.placeholderapp',
+    ],
+    MIDDLEWARE_CLASSES=[
+        'cms.middleware.utils.ApphookReloadMiddleware',
+    ],
+    TEMPLATE_DIRS=(
+        os.path.join(
+            os.path.dirname(__file__),
+            'cms', 'test_utils', 'project', 'templates', 'integration'),
```
Comment (vxsx) [inline testserver.py]: you're likely missing this path, so if tests would involve pages and not admin things would fail

**Comment by vxsx** [comment]
so far looks ok

File: README.rst [hunk 4]
```diff
@@ -51,6 +51,9 @@ There are Django version 1.8 and 1.9 provided.
 To install 1.8 infrastructure please run ``make install18`` and for 1.9 ``make install19``
 
 After this is done you could run tests with following command ``gulp tests``
+Or if you want to run a separate test run ``gulp tests:integration --tests=loginAdmin``
+And if you have to clean the database up from your ghost server run ``--clean`` with your test to clean up the database
```
Comment (vxsx) [inline README.rst]: what's a ghost server o_O

File: djangocms_admin_style/tests/integration/addNewUser.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+        // delete button gets clicked
+        .waitForSelector('#user_form', function () {
+            test.assertExists('#user_form', 'User Form has been loaded');
+            this.click('.deletelink');
+        })
+
+        // confirming that the user gets deleted
+        .waitForSelector('.delete-confirmation', function () {
+            test.assertExists('.delete-confirmation', 'Delete button clicked confirmed');
```
Comment (vxsx) [inline djangocms_admin_style/tests/integration/addNewUser.js]: would be great to have screenshot of delete confirmation i think

File: djangocms_admin_style/tests/integration/addNewUser.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+        // checks if the user has been added to the list
+        .waitForSelector('#changelist-form', function () {
+            test.assertSelectorHasText(
+                '#changelist-form .field-email',
+                globals.user.userEmail,
+                'The User has been updated'
+            );
+
+            phantomcss.screenshot('.change-list', 'user list');
```
Comment (vxsx) [inline djangocms_admin_style/tests/integration/addNewUser.js]: i'm afraid we need bigger area. buttons on top look cutoff

File: README.rst (hunk 4 above)
Comment (lory87) [inline README.rst]: Updated readme

**Comment by vxsx** [comment]
👍

**Comment by vxsx** [comment]
@evildmp would appreciate if you could look into documentation changes

**Comment by vxsx** [comment]
https://github.com/divio/djangocms-casper-helpers is ready to use

File: djangocms_admin_style/tests/integration/addNewUser.js (hunk 5 above)
Comment (vxsx) [inline djangocms_admin_style/tests/integration/addNewUser.js]: resolved

**Comment by lory87** [comment]
👍

**Comment by FinalAngel** [comment]
nice work @vxsx 👍

**Comment by vxsx** [comment]
nice work goes to @lory87 as she did most of it
<3, merging

**Pull request merged**